### PR TITLE
i18n: Mark up GeneralSettings for translations

### LINKED ISSUE #75530: i18n: Mark up `GeneralSettings` for translations
follow instructions [here](https://github.com/grafana/grafana/blob/main/contribute/internationalization.md#how-to-add-a-new-translation-phrase). can verify translation strings have been marked up correctly by running `yarn i18n:pseudo` and using the development-only pseudo locale: 
image

checklist:
- [ ] Form field labels
- [ ] Form field descriptions
- [ ] Form subheadings
- [ ] Delete dashboard button

### PR DESCRIPTION
**Pull request by otilor**
**What is this feature?**

Adds internalization support for `GeneralSettings`

**Why do we need this feature?**

Improve internalization in Grafana.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #75530

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard/components/DashboardSettings/AutoRefreshIntervals.tsx [hunk 1]
```diff
@@ -58,8 +59,11 @@ export const AutoRefreshIntervals = ({
 
   return (
     <Field
-      label="Auto refresh"
-      description="Define the auto refresh intervals that should be available in the auto refresh list."
+      label={t('dashboard-settings.general-settings.auto-refresh.label', 'Auto refresh')}
+      description={t(
+        'dashboard-settings.general-settings.auto-refresh.description',
+        'Define the auto refresh intervals that should be available in the auto refresh list.'
+      )}
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/AutoRefreshIntervals.tsx:66]:
```suggestion
      label={t('dashboard-settings.general.auto-refresh-label', 'Auto refresh')}
      description={t(
        'dashboard-settings.general.auto-refresh-description',
        'Define the auto refresh intervals that should be available in the auto refresh list.'
      )}
```

File: public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx [hunk 2]
```diff
@@ -124,7 +125,10 @@ export function GeneralSettingsUnconnected({
           <Field
             label={
               <HorizontalGroup justify="space-between">
-                <Label htmlFor="title-input">Title</Label>
+                <Label htmlFor="title-input">
+                  <Trans i18nKey="dashboard-settings.general-settings.title.label">Title</Trans>
+                </Label>
+
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx:131]:
```suggestion
                <Label htmlFor="title-input">
                  <Trans i18nKey="dashboard-settings.general.title-label">Title</Trans>
                </Label>

```

File: public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx [hunk 3]
```diff
@@ -141,7 +145,10 @@ export function GeneralSettingsUnconnected({
           <Field
             label={
               <HorizontalGroup justify="space-between">
-                <Label htmlFor="description-input">Description</Label>
+                <Label htmlFor="description-input">
+                  {t('dashboard-settings.general-settings.description.label', 'Description')}
+                </Label>
+
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx:151]:
```suggestion
                <Label htmlFor="description-input">
                  {t('dashboard-settings.general.description-label', 'Description')}
                </Label>

```

File: public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx [hunk 4]
```diff
@@ -155,11 +162,13 @@ export function GeneralSettingsUnconnected({
...
-          <Field label="Tags">
-            <TagsInput id="tags-input" tags={dashboard.tags} onChange={onTagsChange} width={40} />
+          <Field label={t('dashboard-settings.general-settings.tags.label', 'Tags')}>
+            <Trans i18nKey="dashboard-settings.general-settings.tags-input">
+              <TagsInput id="tags-input" tags={dashboard.tags} onChange={onTagsChange} width={40} />
+            </Trans>
           </Field>
 
-          <Field label="Folder">
+          <Field label={t('dashboard-settings.general-settings.folder.label', 'Folder')}>
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx:171]:
```suggestion
          <Field label={t('dashboard-settings.general.folder-label', 'Folder')}>
```

File: public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx [hunk 5]
```diff
@@ -155,11 +162,13 @@ export function GeneralSettingsUnconnected({
               onChange={(e: ChangeEvent<HTMLTextAreaElement>) => onDescriptionChange(e.target.value)}
             />
           </Field>
-          <Field label="Tags">
-            <TagsInput id="tags-input" tags={dashboard.tags} onChange={onTagsChange} width={40} />
+          <Field label={t('dashboard-settings.general-settings.tags.label', 'Tags')}>
+            <Trans i18nKey="dashboard-settings.general-settings.tags-input">
+              <TagsInput id="tags-input" tags={dashboard.tags} onChange={onTagsChange} width={40} />
+            </Trans>
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx:168]: we don't want to translate the `<TagsInput>` component here. if there are translations that need to be done for this component, they need to be done in the `TagsInput` component file itself.
```suggestion
          <Field label={t('dashboard-settings.general.tags-label', 'Tags')}>
            <TagsInput id="tags-input" tags={dashboard.tags} onChange={onTagsChange} width={40} />
```

File: public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx [hunk 6]
```diff
@@ -173,8 +182,11 @@ export function GeneralSettingsUnconnected({
           </Field>
 
           <Field
-            label="Editable"
-            description="Set to read-only to disable all editing. Reload the dashboard for changes to take effect"
+            label={t('dashboard-settings.general-settings.editable.label', 'Editable')}
+            description={t(
+              'dashboard-settings.general-settings.editable-description',
+              'Set to read-only to disable all editing. Reload the dashboard for changes to take effect'
+            )}
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx:189]:
```suggestion
            label={t('dashboard-settings.general.editable-label', 'Editable')}
            description={t(
              'dashboard-settings.general.editable-description',
              'Set to read-only to disable all editing. Reload the dashboard for changes to take effect'
            )}
```

File: public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx [hunk 7]
```diff
@@ -196,10 +208,16 @@ export function GeneralSettingsUnconnected({
         />
 
         {/* @todo: Update "Graph tooltip" description to remove prompt about reloading when resolving #46581 */}
-        <CollapsableSection label="Panel options" isOpen={true}>
+        <CollapsableSection
+          label={t('dashboard-settings.general-settings.panel-options.label', 'Panel options')}
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx:212]:
```suggestion
          label={t('dashboard-settings.general.panel-options-label', 'Panel options')}
```

File: public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx [hunk 8]
```diff
@@ -196,10 +208,16 @@ export function GeneralSettingsUnconnected({
...
+          isOpen={true}
+        >
           <Field
-            label="Graph tooltip"
-            description="Controls tooltip and hover highlight behavior across different panels. Reload the dashboard for changes to take effect"
+            label={t('dashboard-settings.general-settings.panel-options-graph-tooltip.label', 'Graph tooltip')}
+            description={t(
+              'dashboard-settings.general-settings.panel-options-graph-tooltip.description',
+              'Controls tooltip and hover highlight behavior across different panels. Reload the dashboard for changes to take effect'
+            )}
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx:220]:
```suggestion
            label={t('dashboard-settings.general.panel-options-graph-tooltip-label', 'Graph tooltip')}
            description={t(
              'dashboard-settings.general.panel-options-graph-tooltip-description',
              'Controls tooltip and hover highlight behavior across different panels. Reload the dashboard for changes to take effect'
            )}
```

File: public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx [hunk 9]
```diff
@@ -66,8 +67,14 @@ export class TimePickerSettings extends PureComponent<Props, State> {
...
     return (
-      <CollapsableSection label="Time options" isOpen={true}>
-        <Field label="Time zone" data-testid={selectors.components.TimeZonePicker.containerV2}>
+      <CollapsableSection
+        label={t('dashboard-settings.general-settings.time-options-label', 'Time options')}
+        isOpen={true}
+      >
+        <Field
+          label={t('dashboard-settings.general-settings.time-picker-label', 'Time zone')}
+          data-testid={selectors.components.TimeZonePicker.containerV2}
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx:76]:
```suggestion
        label={t('dashboard-settings.time-picker.time-options-label', 'Time options')}
        isOpen={true}
      >
        <Field
          label={t('dashboard-settings.time-picker.time-zone-label', 'Time zone')}
          data-testid={selectors.components.TimeZonePicker.containerV2}
```

File: public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx [hunk 10]
```diff
@@ -76,7 +83,10 @@ export class TimePickerSettings extends PureComponent<Props, State> {
             width={40}
           />
         </Field>
-        <Field label="Week start" data-testid={selectors.components.WeekStartPicker.containerV2}>
+        <Field
+          label={t('dashboard-settings.general-settings.week-start.label', 'Week start')}
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx:87]:
```suggestion
          label={t('dashboard-settings.time-picker.week-start.label', 'Week start')}
```

File: public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx [hunk 11]
```diff
@@ -88,7 +98,13 @@ export class TimePickerSettings extends PureComponent<Props, State> {
...
           onRefreshIntervalChange={this.props.onRefreshIntervalChange}
         />
-        <Field label="Now delay" description="Exclude recent data that may be incomplete.">
+        <Field
+          label={t('dashboard-settings.general-settings.now_delay.label', 'Now delay')}
+          description={t(
+            'dashboard-settings.general-settings.now_delay.description',
+            'Exclude recent data that may be incomplete.'
+          )}
+        >
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx:107]:
```suggestion
        <Field
          label={t('dashboard-settings.time-picker.now-delay-label', 'Now delay')}
          description={t(
            'dashboard-settings.time-picker.now-delay-description',
            'Exclude recent data that may be incomplete.'
          )}
        >
```

File: public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx [hunk 12]
```diff
@@ -97,16 +113,19 @@ export class TimePickerSettings extends PureComponent<Props, State> {
             defaultValue={this.props.nowDelay}
           />
         </Field>
-        <Field label="Hide time picker">
+        <Field label={t('dashboard-settings.general-settings.hide-time-picker', 'Hide time picker')}>
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx:116]:
```suggestion
        <Field label={t('dashboard-settings.time-picker.hide-time-picker', 'Hide time picker')}>
```

File: public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx [hunk 13]
```diff
@@ -97,16 +113,19 @@ export class TimePickerSettings extends PureComponent<Props, State> {
...
           />
         </Field>
         <Field
-          label="Refresh live dashboards"
-          description="Continuously re-draw panels where the time range references 'now'"
+          label={t('dashboard-settings.general-settings.refresh-live-dashboards.label', 'Refresh live dashboards')}
+          description={t(
+            'dashboard-settings.general-settings.refresh-live-dashboards.description',
+            "Continuously re-draw panels where the time range references 'now'"
+          )}
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx:128]:
```suggestion
          label={t('dashboard-settings.time-picker.refresh-live-dashboards-label', 'Refresh live dashboards')}
          description={t(
            'dashboard-settings.time-picker.refresh-live-dashboards-description',
            "Continuously re-draw panels where the time range references 'now'"
          )}
```

File: public/app/features/dashboard/components/DeleteDashboard/DeleteDashboardButton.tsx [hunk 14]
```diff
@@ -23,7 +24,7 @@ export const DeleteDashboardButton = ({ dashboard }: Props) => (
         }}
         aria-label="Dashboard settings page delete dashboard button"
       >
-        Delete Dashboard
+        <Trans i18nKey="dashboard-delete-button">Delete Dashboard</Trans>
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DeleteDashboard/DeleteDashboardButton.tsx:27]:
```suggestion
        <Trans i18nKey="dashboard-settings.dashboard-delete-button">Delete Dashboard</Trans>
```

**Review by ashharrison90 — COMMENTED** [review]
mainly just nitpicks on naming, looks great otherwise 🙌

**Comment by otilor** [comment]
> mainly just nitpicks on naming, looks great otherwise 🙌

I've updated it

**Review by ashharrison90 — COMMENTED** [review]
@otilor one more to change, then done 👍

File: public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx [hunk 15]
```diff
@@ -76,7 +83,10 @@ export class TimePickerSettings extends PureComponent<Props, State> {
             width={40}
           />
         </Field>
-        <Field label="Week start" data-testid={selectors.components.WeekStartPicker.containerV2}>
+        <Field
+          label={t('dashboard-settings.time-picker.week-start.label', 'Week start')}
```
Comment (ashharrison90) [inline public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx:87]:
```suggestion
          label={t('dashboard-settings.time-picker.week-start-label', 'Week start')}
```

sorry, missed one here on my review. the reason for these changes is we only want to use up to 3 segments as part of the key name. see our naming convention guidelines [here](https://github.com/grafana/grafana/blob/main/contribute/internationalization.md#phrase-id-naming-convention)

**Review by ashharrison90 — APPROVED** [review]
looks great! 🙌 

thanks for the contribution @otilor 🥳

**Pull request merged**